riscv-page-tables/src/pte.rs

```diff
@@ -85,6 +85,13 @@ impl Pte {
         self.0 = (pfn.bits() << PFN_SHIFT) | status.bits | PteFieldBit::Valid.mask();
     }
 
+    /// Updates the pfn part of the entry, keeping everything else same. Returns the old pfn value.
+    pub fn update_pfn(&mut self, pfn: SupervisorPfn) -> SupervisorPfn {
```

Comment:
This also seems unsafe, @abrestic-rivos is the 'set' function above still safe?

Reply:
Fixed.